### PR TITLE
Add unenrol commands and tests

### PR DESCRIPTION
**Pull request by MReyna12**

### CODE REVIEW

**Review by rnathuji — APPROVED** [review]
@MReyna12 - I just had a couple of minor comments here which should be quick to fix. I'm going to go ahead and approve.

File: tests/test_main.py [hunk 1]
```diff
@@ -181,6 +181,38 @@ def test_enrol_bulk_csv(requests_mock, tmp_path):
...
+            writer.writerows([{utils.CSV_USER_EMAIL: 'tommichaels@gmail.com'}])
+
+        result = runner.invoke(cli, ['unenrol-bulk',
+                                     '2', 'unenrolstudents.csv'],
+                               env=TEST_ENV)
+
+        assert result.exit_code == 0
+
+
+def test_unenrol_bulk_csv(requests_mock, tmp_path):
```
Comment (rnathuji) [inline tests/test_main.py:203]: Minor: When you remove the use of the fixture below, `requests_mock` can be removed here too.

File: moodlecli/main.py [hunk 2]
```diff
@@ -199,6 +199,34 @@ def enrol_bulk(course_id, role_shortname, userdata_csv):
...
+        utils.unenrol_bulk_input_csv_fieldnames()
+    )
+    writer.writeheader()
+
+
+@cli.command()
+@click.argument('course_id')
+@click.argument('userdata_csv', type=click.File(mode='r'))
+def unenrol_bulk(course_id, userdata_csv):
+    """Bulk unenrol users from course with role"""
```
[the remark this answers is not in the document]
Comment (MReyna12) [reply moodlecli/main.py:217]: Fixed:

image

File: tests/test_main.py [hunk 3]
```diff
@@ -181,6 +181,38 @@ def test_enrol_bulk_csv(requests_mock, tmp_path):
...
+    test_json = {'foo': 'bar'}
+    runner = CliRunner()
+    with runner.isolated_filesystem(temp_dir=tmp_path):
+        requests_mock.get(f'{TEST_MOODLE_URL}{moodle.MOODLE_WEBSERVICE_PATH}',
+                          json=test_json)
```
[the remark this answers is not in the document]
Comment (MReyna12) [reply tests/test_main.py:208]: Okay, I've removed them all.

File: tests/test_main.py (hunk 1 above)
Comment (MReyna12) [reply tests/test_main.py:203]: `requests_mock` has been removed.

**Pull request merged**